### PR TITLE
chore: address vanity nits in 08-wasm (code structure, godocs)

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Moves `MockEngine` struct def further up in `mock_engine.go` file for better readability/clarity
- Consolidate `sdk.Msg` codec registration to leverage varargs argument
- Add godocs to various exported symbols (`WasmConfig`, `Store` wrapper)
- Move store code handling checksums state to `store.go`, delete ambiguous `wasm.go` and `wasm_test.go` files.

I think there are more improvements that could be made to help structure the code and package layout more optimally but we can follow up on them in future.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by crodriguezvega — APPROVED** [review]
💅 🙏

File: modules/light-clients/08-wasm/types/store.go [hunk 1]
```diff
@@ -205,22 +253,27 @@ func newStoreAdapter(s storetypes.KVStore) *storeAdapter {
 	return &storeAdapter{parent: s}
 }
 
+// Get implements the KVStore interface.
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/light-clients/08-wasm/types/store.go]: ah, its a reduced interface! nice catch :)

File: modules/light-clients/08-wasm/testing/mock_engine.go [hunk 2]
```diff
@@ -34,6 +34,27 @@ type (
...
+
+	// queryCallbacks contains a mapping of queryMsg field type name to callback function.
+	queryCallbacks map[string]queryFn
+	sudoCallbacks  map[string]sudoFn
+
+	// contracts contains a mapping of checksum to code.
+	storedContracts map[uint32][]byte
+}
+
+// NewMockWasmEngine creates and returns a new instance of the mock wasmvm for testing purposes.
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/light-clients/08-wasm/testing/mock_engine.go:57]: much better now I think! 👍🏻

File: modules/light-clients/08-wasm/testing/mock_engine.go [hunk 3]
```diff
@@ -34,6 +34,31 @@ type (
 	sudoFn func(checksum wasmvm.Checksum, env wasmvmtypes.Env, sudoMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error)
 )
 
+// MockWasmEngine implements types.WasmEngine for testing purpose. One or multiple messages can be stubbed.
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/testing/mock_engine.go:37]:
```suggestion
// MockWasmEngine implements types.WasmEngine for testing purposes. One or multiple messages can be stubbed.
```

**Pull request merged**